### PR TITLE
Add Setting for Default Font on New Plots

### LINKED ISSUE #28597: WB Plot Fonts
**Original reporter:** SANS group / Rob Dalgliesh

## Feature Request
Ability to alter the font used on plots

I think FontType would acceptably be a settable default in File>Settings>Plots

[Worth discussing with others in the team: Should a font size for text on plots be an option or is this harder to do / maybe not always going to be what the user wants?]

### PR DESCRIPTION
**Pull request by cailafinn**
**Description of work.**
Adds some settings that change the `rcParams` font elements of matplotlib so that plots use a font selected by the user. The font chosen is saved to the ConfigService and `mantid.user.properties` file. 

**To test:**
1. Workbench
2.  Plot something.
3. Make a note of the font used for labels and legends
4. File -> Settings -> Plots
5. Change the default font
6. Plot something

Fixes #28597 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by StephenSmith25 — COMMENTED** [review]
This works well and code changes look good. Just one question about the default value when you load up the settings for the first time. For me it seems to arbitrarily point to the first font in the list, rather than the current font / default font used by Matplotlib. Is there a reason we can't set the default font within the properties file?

**Comment by ConorMFinn** [comment]
> This works well and code changes look good. Just one question about the default value when you load up the settings for the first time. For me it seems to arbitrarily point to the first font in the list, rather than the current font / default font used by Matplotlib. Is there a reason we can't set the default font within the properties file?

It should be taking the value from the matplotlib `rcParams` if nothing is in the properties file. It does seem to work on my local build. I suppose I could add the font used to the properties template but I'd be hesitant in case matplotlib change the defaults. What do you think?

**Comment by StephenSmith25** [comment]
> 
> 
> > This works well and code changes look good. Just one question about the default value when you load up the settings for the first time. For me it seems to arbitrarily point to the first font in the list, rather than the current font / default font used by Matplotlib. Is there a reason we can't set the default font within the properties file?
> 
> It should be taking the value from the matplotlib `rcParams` if nothing is in the properties file. It does seem to work on my local build. I suppose I could add the font used to the properties template but I'd be hesitant in case matplotlib change the defaults. What do you think?

Yeah I agree, it would be bad to hard code the default value. I think my issue is that the `DejaVu Sans` font isn't appearing in my list, so when it tries to set the default using `setCurrentText` it does nothing as the font isn't in the list. I think this is most likely a problem on my end. It doesn't stop anything from working, just the font in the settings box doesn't correspond to what’s being used. I wonder why that font is missing for me..

**Review by StephenSmith25 — COMMENTED** [review]
Okay so the box is now full when I first open it, but if I plot data and then go back to the plot setting the box is empty. Wondering if this is isolated to my PC.

**Review by StephenSmith25 — DISMISSED** [review]
This now works all works for me, good work. I have one small reservation which I do not believe is worth holding up the PR for. The font list is very large and some of the fonts don't support some of the symbols we use in the plots (e.g. a few of them are missing the minus sign, so a blank box appears where the minus sign should be). I guess this is up to the user to choose a suitable font, as it is in Microsoft Word for example.

File: docs/source/concepts/PropertiesFile.rst [hunk 1]
```diff
@@ -262,6 +262,8 @@ Plotting Settings
 +---------------------------------+------------------------------------------------------------------+---------------------+
 |``plots.ShowMinorGridlines``     |Whether to show minor gridlines on plots                          | ``On``, ``Off``     |
 +---------------------------------+------------------------------------------------------------------+---------------------+
+|``plots.font``                   |The default font for labels and titles on plots.                  |``Helvetica``        |
```
Comment (DavidFair) [inline docs/source/concepts/PropertiesFile.rst:265]: Our docs show that the default font is Helvetica, but the template properties is blank - is this being set somewhere else then or is it the MPL default?

File: qt/applications/workbench/workbench/widgets/settings/plots/presenter.py [hunk 2]
```diff
@@ -230,6 +239,32 @@ def action_default_colormap_changed(self):
...
+            fonts.add(font_name)
+
+        if mpl.rcParams['font.family'][0] in ['sans-serif', 'serif', 'cursive', 'fantasy', 'monospace']:
+            current_mpl_font = mpl.rcParams['font.' + mpl.rcParams['font.family'][0]][0]
+        else:
+            current_mpl_font = mpl.rcParams['font.family'][0]
+        fonts.add(current_mpl_font)
+        fonts = sorted(fonts)
+
+        self.view.plot_font.addItems(fonts)
```
Comment (DavidFair) [inline qt/applications/workbench/workbench/widgets/settings/plots/presenter.py:260]: I'm a little concerned this presenter is starting to grow into a model-presenter combo:
Everything from 242-259 should be in a model, and then the presenter can focus on just the view

**Review by DavidFair — CHANGES_REQUESTED** [review]

File: docs/source/concepts/PropertiesFile.rst (hunk 1 above)
Comment (StephenSmith25) [reply docs/source/concepts/PropertiesFile.rst:265]: The last column in this table refers to an example value not a default value.

**Review by DavidFair — APPROVED** [review]
Thanks for making the change, I've just retested and this still works correctly for me

**Pull request merged**